widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.kt

```diff
@@ -176,6 +179,24 @@ internal class ChatController(
         chatState = ChatState()
         subscribeToEngagement()
         decideOnQueueingUseCase().unSafeSubscribe(::enqueueForEngagement)
+        screenSharingUseCase().unSafeSubscribe(::handleScreenSharingState)
+    }
+
+    private fun handleScreenSharingState(screenSharingState: ScreenSharingState) {
```

Comment:
Can we extract repeated code?

Reply:
I don't think so.
Even though we process this state in all places inside similar `when` blocks, the functions we call based on every state are different.